aliasmgr/aliasmgr.go

```diff
@@ -404,16 +454,51 @@ func (m *Manager) DeleteSixConfs(baseScid lnwire.ShortChannelID) error {
 
 		var baseBytes [8]byte
 		byteOrder.PutUint64(baseBytes[:], baseScid.ToUint64())
-		return baseConfBucket.Put(baseBytes[:], []byte{})
-	}, func() {})
+		err = baseConfBucket.Put(baseBytes[:], []byte{})
+		if err != nil {
+			return err
+		}
+
+		// Check which aliases for this base are marked as persistent.
+		aliasToBaseBucket, err := tx.CreateTopLevelBucket(aliasBucket)
+		if err != nil {
+			return err
+		}
+
+		return aliasToBaseBucket.ForEach(func(k, v []byte) error {
```

Comment:
Rather than iterate over this entire bucket, couldn't we do a random read here?

Reply:
Wdym by random read? We need to check all entries here and see which of the aliases correspond to the confirmed base scid.

While I was reading this comment I realized we were iterating twice over this set. So with the new commit I do everything in a single pass. PTAL